jina/types/arrays/neural_ops.py

```diff
@@ -250,11 +249,28 @@ def visualize(
             plt.show()
 
     @property
-    def embeddings(self, indices: Optional[slice] = None) -> np.ndarray:
+    def embeddings(self) -> np.ndarray:
         """Return a `np.ndarray` stacking all the `embedding` attributes as rows.
-        If indices is passed the embeddings from the indices are retrieved.
 
-        Example: `self.embeddings[10:20]` will return 10 embeddings from positions 10 to 20
+        Warning: This operation assumes all embeddings have the same shape.
+
+        Warning: This operation currently does not support sparse arrays.
+
+        :return: embeddings stacked per row as `np.ndarray`.
+        """
+
+        x_mat = b''.join(d.proto.embedding.dense.buffer for d in self)
```

Comment:
I would recommend calling `self._get_embeddings(None)`?

Reply:
It is a sensible recomendation, but this non-for loop version is a bit more efficient. I just don´t do it in the other case because I can´t (since there are slices involved)